File: demo/internal/model/Branch.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Branch Declaration
//==============================================================
type Branch struct {
    gorm.Model
     Name                                    string
    BranchCode                                    string
    Address                                                            string
    Phone                                    string
    OpeningHours                                    string
    BankId         *uint
    Bank           *Bank `gorm:"foreignKey:BankId"`
     Accounts           []Account `gorm:"foreignKey:AccountsFromBranchId"`
     LoanAccounts           []LoanAccount `gorm:"foreignKey:LoanAccountsFromBranchId"`
     Atms           []ATM `gorm:"foreignKey:AtmsFromBranchId"`

// parent associations as their child

}

File: bankingonaspdotnet/Domain/Branch.cs

using bankingonaspdotnet.Contracts;

namespace bankingonaspdotnet.Domain;

public class Branch
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long? BranchId { get; set; } 
 public virtual string? Name { get; set; } 
 public virtual string? BranchCode { get; set; } 
 public virtual Address? Address { get; set; } 
 public virtual string? Phone { get; set; } 
 public virtual string? OpeningHours { get; set; } 
public virtual Bank? Bank { get; set; } 
public virtual ICollection<Account> Accounts { get; set; } = new List<Account>();
public virtual ICollection<LoanAccount> LoanAccounts { get; set; } = new List<LoanAccount>();
public virtual ICollection<ATM> Atms { get; set; } = new List<ATM>();

    public static Branch FromRequest(BranchRequest request) {
        return new Branch {
            Id = request.Id,
            Name = request.Name,
            BranchCode = request.BranchCode,
            Address = request.Address,
            Phone = request.Phone,
            OpeningHours = request.OpeningHours,
        };
    }
}
